src/main/java/com/dubture/jenkins/digitalocean/DigitalOcean.java

```diff
@@ -88,17 +96,24 @@ static List<Size> getAvailableSizes(String authToken) throws DigitalOceanExcepti
      * @throws DigitalOceanException
      * @throws RequestUnsuccessfulException
      */
-    static SortedMap<String,Image> getAvailableImages(String authToken) throws DigitalOceanException, RequestUnsuccessfulException {
+    static SortedMap<String,Image> getAvailableImages(String authToken, ImageFilter filter) throws DigitalOceanException, RequestUnsuccessfulException {
         DigitalOceanClient client = new DigitalOceanClient(authToken);
 
         SortedMap<String,Image> availableImages = new TreeMap<>(ignoringCase());
 
-        Images images;
+        Images images = null;
         int page = 0;
 
         do {
             page += 1;
-            images = client.getAvailableImages(page, Integer.MAX_VALUE);
+            switch (filter) {
+                case ALLIMAGES:
```

Comment:
What happens if filter isn't one of these two? Does jenkins throw a 500? does it reset to ALLIMAGES?